packages/lesswrong/components/ultraFeed/UltraFeedSettings.tsx

```diff
@@ -0,0 +1,411 @@
+    },
+  },
+  invalidInput: {
+    borderColor: theme.palette.error.main,
+  },
+  errorMessage: {
+    color: theme.palette.error.main,
+    fontSize: '0.8rem',
+    marginTop: 4,
+  },
+  buttonDisabled: {
+    backgroundColor: theme.palette.grey[400],
+    color: theme.palette.grey[800],
+    cursor: 'not-allowed',
+    '&:hover': {
+      backgroundColor: theme.palette.grey[400],
+    },
+  },
+  disabled: {
+    opacity: 0.5,
+    pointerEvents: 'none',
+  },
+}));
+
+const UltraFeedSettings = ({ onClose }: { onClose: () => void }) => {
+  const classes = useStyles(styles);
+  const { settings, updateSetting, resetSettings } = useUltraFeedSettings();
+  
+  // Local state for form values
+  const [formValues, setFormValues] = useState({
+    postTruncationBreakpoints: [...settings.postTruncationBreakpoints],
+    commentTruncationBreakpoints: [...settings.commentTruncationBreakpoints],
+    collapsedCommentTruncation: settings.collapsedCommentTruncation,
+    lineClampNumberOfLines: settings.lineClampNumberOfLines,
+  });
+  
+  // Track validation errors
+  const [errors, setErrors] = useState({
+    postTruncationBreakpoints: [] as number[],
+    commentTruncationBreakpoints: [] as number[],
+    collapsedCommentTruncation: false,
+    lineClampNumberOfLines: false,
+  });
+  
+  // Update array of numbers at specific index
+  const handleArrayChange = useCallback((
+    key: 'postTruncationBreakpoints' | 'commentTruncationBreakpoints',
+    index: number,
+    value: string
+  ) => {
+    setFormValues(prev => {
+      const newArray = [...prev[key]];
+      
+      if (value === '') {
+        // Allow empty string but mark as error
+        newArray[index] = '' as any; // Using 'any' to allow empty string temporarily
+        
+        // Update errors
+        setErrors(prevErrors => {
+          const newErrors = [...prevErrors[key]];
+          if (!newErrors.includes(index)) {
+            newErrors.push(index);
+          }
+          return {
+            ...prevErrors,
+            [key]: newErrors
+          };
+        });
+      } else {
+        const numValue = parseInt(value, 10);
+        newArray[index] = numValue;
+        
+        // Clear error if value is valid
+        if (!isNaN(numValue)) {
+          setErrors(prevErrors => {
+            const newErrors = [...prevErrors[key]].filter(i => i !== index);
+            return {
+              ...prevErrors,
+              [key]: newErrors
+            };
+          });
+        }
+      }
+      
+      return {
+        ...prev,
+        [key]: newArray
+      };
+    });
+  }, []);
+
+  // Handle number input change
+  const handleNumberChange = useCallback((
+    key: 'collapsedCommentTruncation' | 'lineClampNumberOfLines',
+    value: string
+  ) => {
+    if (value === '') {
+      // Allow empty input but mark as error
+      setFormValues(prev => ({
+        ...prev,
+        [key]: '' as any // Using 'any' to allow empty string temporarily
```

Comment:
idk what's going on here but I don't like it

Reply:
addressed